src/sspmprk.jl

```diff
@@ -0,0 +1,908 @@
+"""
+    SSPMPRK22(α, β; [linsolve = ...])
+
+A family of second-order modified Patankar-Runge-Kutta algorithms for 
+production-destruction systems. Each member of this family is a one-step, two-stage method which is
+second-order accurate, unconditionally positivity-preserving, and linearly
+implicit. The parameters `α` and `β` are described by Huang and Shu (2019) and
+studied by Huang, Izgin, Kopecz, Meister and Shu (2023). 
+The difference to [`MPRK22`](@ref) is that this method is based on the SSP formulation of 
+an explicit second-order Runge-Kutta method. This family of schemes contains the [`MPRK22`](@ref) family,
+where `MPRK22(α) = SSMPRK22(0, α)` applies.
+
+The scheme was introduced by Huang and Shu for conservative production-destruction systems. 
+For nonconservative production–destruction systems we use the straight forward extension
+analogous to [`MPE`](@ref).
+
+This modified Patankar-Runge-Kutta method requires the special structure of a
+[`PDSProblem`](@ref) or a [`ConservativePDSProblem`](@ref).
+
+You can optionally choose the linear solver to be used by passing an
+algorithm from [LinearSolve.jl](https://github.com/SciML/LinearSolve.jl)
+as keyword argument `linsolve`.
+You can also choose the parameter`small_constant` which is added to all Patankar-weight denominators 
+to avoid divisions by zero.
+
+## References
+
+- Juntao Huang and Chi-Wang Shu.
+  "Positivity-Preserving Time Discretizations for Production–Destruction Equations 
+  with Applications to Non-equilibrium Flows."
+  Journal of Scientific Computing 78 (2019): 1811–1839
+  [DOI: 10.1007/s10915-018-0852-1](https://doi.org/10.1007/s10915-018-0852-1)
+- Juntao Huang, Thomas Izgin, Stefan Kopecz, Andreas Meister and Chi-Wang Shu.
+  "On the stability of strong-stability-preserving modified Patankar-Runge-Kutta schemes."
+  ESAIM: Mathematical Modelling and Numerical Analysis 57 (2023):1063–1086
+  [DOI: 10.1051/m2an/2023005](https://doi.org/10.1051/m2an/2023005)
+"""
+struct SSPMPRK22{T, F} <: OrdinaryDiffEqAdaptiveAlgorithm
+    alpha::T
+    beta::T
+    linsolve::F
+    small_constant::T
+end
+
+function SSPMPRK22(alpha, beta; linsolve = LUFactorization(),
+                   small_constant = floatmin(alpha))
+    SSPMPRK22{typeof(alpha), typeof(linsolve)}(alpha, beta, linsolve, small_constant)
+end
+
+alg_order(::SSPMPRK22) = 2
+isfsal(::SSPMPRK22) = false
+
+function get_constant_parameters(alg::SSPMPRK22)
+    if !((0 ≤ alg.alpha ≤ 1) && (alg.beta ≥ 0) &&
+         (alg.alpha * alg.beta + 1 / (2 * alg.beta) ≤ 1))
+        throw(ArgumentError("SSPMPRK22 requires 0 ≤ α ≤ 1, β ≥ 0 and αβ + 1/(2β) ≤ 1."))
+    end
+
+    a21 = alg.alpha
+    a10 = one(alg.alpha)
+    a20 = 1 - a21
+
+    b10 = alg.beta
+    b20 = 1 - 1 / (2 * b10) - a21 * b10
+    b21 = 1 / (2 * b10)
+
+    s = (b20 + b21 + a21 * b10^2) / (b10 * (b20 + b21))
+
+    # This should never happen
+    if !all((a21, a10, a20, b10, b20, b21) .≥ 0)
```

Comment:
```suggestion
    if any(<(0), (a21, a10, a20, b10, b20, b21))
```
This does not need to construct a temporary tuple and can return early if a coefficient is negative.